File: SourceAFIS/Matching/BestMatchSkipper.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;

namespace SourceAFIS.Simple
{
    class BestMatchSkipper
    {
        float[][] Collected;

        public struct PersonsSkipScore
        {
            public int Person;
            public float Score;
        }

        public BestMatchSkipper(int persons, int skip)
        {
            Collected = new float[skip + 1][];
            for (int i = 0; i < Collected.Length; ++i)
            {
                Collected[i] = new float[persons];
                for (int j = 0; j < Collected[i].Length; ++j)
                    Collected[i][j] = -1;
            }
        }

        public void AddScore(int person, float score)
        {
            for (int nth = Collected.Length - 1; nth >= 0; --nth)
            {
                if (Collected[nth][person] < score)
                {
                    if (nth + 1 < Collected.Length)
                        Collected[nth + 1][person] = Collected[nth][person];
                    Collected[nth][person] = score;
                }
            }
        }

        public float GetSkipScore(int person)
        {
            float score = 0;
            for (int nth = Collected.Length - 1; nth >= 0; --nth)
                if (Collected[nth][person] > 0)
                {
                    score = Collected[nth][person];
                    break;
                }
            return score;
        }

        public PersonsSkipScore[] GetSortedScores()
        {
            PersonsSkipScore[] results = new PersonsSkipScore[Collected[0].Length];
            for (int person = 0; person < results.Length; ++person)
            {
                results[person].Person = person;
                results[person].Score = GetSkipScore(person);
            }
            Array.Sort(results, (left, right) => Calc.Compare(right.Score, left.Score));
            return results;
        }
    }
}

File: java/sourceafis/src/sourceafis/matching/BestMatchSkipper.java
package sourceafis.matching;

import java.util.Arrays;
import java.util.Comparator;
import sourceafis.general.Calc;

public class BestMatchSkipper
{
    float[][] collected;
    public class PersonsSkipScore{
    	public int person;
    	public float score;
    }
    
    public BestMatchSkipper(int persons, int skip)
    {
        collected = new float[skip + 1][];
        for (int i = 0; i < collected.length; ++i)
        {
            collected[i] = new float[persons];
            for (int j = 0; j < collected[i].length; ++j)
                collected[i][j] = -1;
        }
    }

    public void addScore(int person, float score)
    {
        for (int nth = collected.length - 1; nth >= 0; --nth)
        {
            if (collected[nth][person] < score)
            {
                if (nth + 1 < collected.length)
                    collected[nth + 1][person] = collected[nth][person];
                collected[nth][person] = score;
            }
        }
    }

    public float getSkipScore(int person)
    {
        float score = 0;
        for (int nth = collected.length - 1; nth >= 0; --nth)
            if (collected[nth][person] > 0)
            {
                score = collected[nth][person];
                break;
            }
        return score;
    }
   /*
    * 
    */
    public PersonsSkipScore[] getSortedScores()
    {
        PersonsSkipScore[] results = new PersonsSkipScore[collected[0].length];
        for (int person = 0; person < results.length; ++person)
        {   results[person]=new PersonsSkipScore();
            results[person].person = person;
            results[person].score = getSkipScore(person);
        }
        
          Arrays.sort(results,new Comparator<PersonsSkipScore>() {
			public int compare(PersonsSkipScore left, PersonsSkipScore right) {
			  return	Calc.Compare(right.score, left.score);
		      }
		   });
        return results;
    }
}